src/xg.cpp

```diff
@@ -2206,6 +2206,145 @@ int64_t XG::where_to(int64_t current_side, int64_t visit_offset, int64_t new_sid
     return new_visit_offset;
 }
 
+int64_t XG::node_height(XG::ThreadMapping node) const {
+  return h_iv[(node_rank_as_entity(node.node_id) - 1) * 2 + node.is_reverse];
+}
+
+int64_t XG::where_to(int64_t current_side, int64_t visit_offset, int64_t new_side, vector<Edge>& edges, vector<Edge>& edges_out) const {
```

Comment:
Can this where_to share code with the original at all somehow? Maybe the old one can just wrap calls to this one? Or would that be too much overhead?